libnmstate/netapplier.py

```diff
@@ -59,6 +59,7 @@ def apply(desired_state, verify_change=True, commit=True, rollback_timeout=60):
     desired_state = copy.deepcopy(desired_state)
     validator.validate(desired_state)
     validator.validate_capabilities(desired_state, netinfo.capabilities())
+    validator.validate_interface_unique_name(desired_state)
```

Comment:
IMHO this commit contradicts the other one, since this shows that multiple interfaces with the same name are not supported while the other one indicates that they will be supported eventually.

Reply:
It will help if you suggest an alternative.
The reason it is not in one commit is to preserve the original contribution.

Reply:
Not sure, what you would like me to suggest as an alternative. The easiest one would be not to support multiple interfaces with the same name and drop the first commit. If multiple interfaces are going to be supported, it should probably raise an `NmstateNotImplementedError` for the cases it is going to be supported.

Reply:
The original change was contributed to expose the problem through a test. The test fails in an ugly manner (not at the validation but at the verification, this was explained in the solution commit). It was marked as xfail because we wanted to fix it and raise a proper error and earlier with enough info (this was included in the fix commit message was well).

The second commit fixes and removes the xfail with a specific exception at the validation failure.

I want to keep the original contribution intact with just fixes.